cost/aws/superseded_instance/aws_superseded_instance_remediation.pt

```diff
@@ -0,0 +1,217 @@
+  description "Email addresses of the recipients you wish to notify"
+end
+
+parameter "param_old_instance_type" do
+  label "Disallowed Instance Type"
+  type "string"
+  description "Old Instance Type"
+end
+
+parameter "param_new_instance_type" do
+  label "Superseded Instance Type"
+  type "string"
+  description "New Instance Type"
+end
+
+parameter "param_exclude_tags" do
+  label "Exclude Tags"
+  description "list of excluded servers"
+  type "list"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+auth "auth_rs", type: "rightscale"
+
+###############################################################################
+# Resources
+###############################################################################
+
+resources "clouds", type: "rs_cm.clouds" do
+  filter do
+    cloud_type "amazon"
+  end
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_instances_16" do
+  iterate(@clouds)
+  request do
+    run_script $js_instances_16, rs_cm_host, href(iter_item), rs_project_id, $param_exclude_tags
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[*]") do
+      field "href", jmes_path(col_item, "href")
+      field "name", jmes_path(col_item, "name")
+      field "description", jmes_path(col_item, "description")
+      field "legacy_id", to_n(jmes_path(col_item, "legacy_id"))
+      field "resource_uid", jmes_path(col_item, "resource_uid")
+      field "instance_type", jmes_path(col_item, "links.instance_type.name")
+      field "region", jmes_path(col_item, "links.cloud.region")
+      field "cloud_name", val(iter_item, "display_name")
+      field "locked", jmes_path(col_item, "locked")
+      field "state", jmes_path(col_item, "state")
+    end
+  end
+end
+
+datasource "ds_combined_instance_data" do
+  run_script $js_merge_instance_data, $ds_instances_16, $param_old_instance_type
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_instances_16", type: "javascript" do
+  result "request"
+  parameters "rs_cm_host","cloud_href","rs_project_id","param_exclude_tags"
+  code <<-EOS
+    var filtered_tags = ""
+    if ( param_exclude_tags.length > 0 ){
+      var filtered_tags = "&tag!=" + param_exclude_tags.join(",");
+    }
+
+    request = {
+      "auth": "auth_rs",
+      "verb": "GET",
+      "host": rs_cm_host,
+      "path": cloud_href + "/instances",
+      "headers": {
+        "X-Api-Version": "1.6",
+        "X-Account": \""+rs_project_id+\""
+      },
+      "query_params": {
+        "view": "full",
+        "filter": "state=provisioned,operational" + filtered_tags
```

Comment:
what about state=running?  isn't provisioned what raw instance report for stopped?